clang/test/dpct/device_virtual_function.cu

```diff
@@ -0,0 +1,60 @@
+// RUN: dpct --format-range=none -out-root %T/device_virtual_function %s --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
+// RUN: FileCheck %s --match-full-lines --input-file %T/device_virtual_function/device_virtual_function.dp.cpp
+
+#include <cuda_runtime.h>
+template <typename T>
+class TestVirtual {
+public:
+
+    __device__ TestVirtual() {}
+// CHECK: /*
+// CHECK-NEXT: DPCT1109:{{[0-9]+}}: Virtual functions cannot be called in a SYCL kernel or by functions called by the kernel. You may need to adjust the code.
+// CHECK-NEXT: */
+    __device__ virtual ~TestVirtual() {}
+// CHECK: /*
+// CHECK-NEXT: DPCT1109:{{[0-9]+}}: Virtual functions cannot be called in a SYCL kernel or by functions called by the kernel. You may need to adjust the code.
+// CHECK-NEXT: */
+    __device__ virtual void push(const T &&e)= 0; 
+};
+template <typename T>
+class TestSeqContainer : public TestVirtual<T> {
+public:
+    __device__ TestSeqContainer(int size) : index_top(-1) { m_data = new T[size]; }
+
+    __device__ ~TestSeqContainer() {
+        if (m_data) delete []m_data;
+    }
+    // CHECK: /*
+    // CHECK-NEXT: DPCT1109:{{[0-9]+}}: Virtual functions cannot be called in a SYCL kernel or by functions called by the kernel. You may need to adjust the code.
+    // CHECK-NEXT: */
+    __device__ virtual void push(const T &&e) {
+        if (m_data) {
+           int idx = atomicAdd(&this->index_top, 1);
+           m_data[idx] = e;
+        }
+    }
+private:
+    T *m_data;
+    int index_top;
+
+};
+__global__ void func(){
+    
+    auto seq = new TestSeqContainer<int>(10);
+    seq->push(10);
```

Comment:
for push(),  will it be easier to understand if emit DPCT1109 also in Line 44 ( it is a call point)?